http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BackgroundProcess.scala

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.Show
+import cats.effect.Async
+import cats.syntax.all._
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.typelevel.log4cats.StructuredLogger
+
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Defines a common interface that certain background processes should implement.
+ * This type of background processes do the following periodically:
+ * 1. Query the database for a list of candidates
+ * 2. Filter the candidates based on some criteria
+ * 3. Perform some action on the filtered candidates
+ */
+trait BackgroundProcess[F[_], A] {
```

Comment:
For name, maybe LeonardoResourceMonitor? There are a number of other *Monitors around the system, which seem to work as long-running continuous jobs.

Reply:
the entire repo is `leonardo`, I don't quite understand the rational of adding `leonardo` in front of things...`Resource` is quite overloaded too

Reply:
Hence `LeonardoResource`... I don't think there's a problem with being specific here, is there? Compare the `WorkspaceManagerResourceMonitor`

Reply:
What do Runtimes and Apps have in common? They're both cloud environments - would it be better to call this a `CloudEnvironmentMonitor`?

Reply:
We don't typically call apps as `CloudEnvironments`

Reply:
The UI does - what are they in our code, though? That's why I suggested "Resources," because I think that's how Leo refers to runtimes, apps, disks, etc - they're Leo resources. I'd love a better word, if there is one

Reply:
How about we leave it as is until a better name emerges